src/components/Tabs/Tabs.tsx

```diff
@@ -167,13 +121,13 @@ export const Tabs = ({
    * Autogenerate ids on tabs if not defined.
    */
   useEffect(() => {
-    // TODO: improve `any` type
     setId(
-      tabs().map((tab: any) => (tab.props.id ? tab.props.id : getUID(tab))),
+      tabs().map((tab: React.ReactElement<TabProps>) =>
```

Comment:
Can we update the types of `tabs()` so that TS knows what these are without us having to specify it?

I'm guessing we don't have good types for `react-children-by-type`, so there's 2 options I see:
1. Remove `react-children-by-type` and do whatever logic we need ourselves
2. Write better custom types for `react-children-by-type`
3. Pass a type param to `useCallback`

3 is the easiest. Would have a similar result to specifying here (since either way we're telling TS what to expect). But I think it's better since generally we shouldn't need to specify the types of callback args.

Reply:
Can also do this later.